### DISCUSSION TITLE
A way to make specific children of scenes visible when instantiated, as an alternative to making all of them visible with editable children

### DISCUSSION THREAD

**Discussion by Teltt** [godotengine/godot-proposals#9312] [Editor]
Hello, often when I am making a scene I need to put the most important script at the top, or use editable children that risks data loss.

I find myself wanting to adjust the data of a few specific nodes of the child scene without adding an export property to the parent, or using editable children to expose unnecessary nodes that I do not want to touch.

use cases:
Reusable area2d scene, that can change it's collision shape. or be part of a parent scene with a model or physics body, that does not need to be duplicated for each variant.
A scene that can modify it's visual look based on a random seed, parented by a script that does other things for the object.
Scenes with heavily parameterized components, so that I don't have to expose them to the parents nodes for level gimmicks and such, and instead edit the exports directly, like lower gravity

**Comment by Teltt** [comment]
another use case is to expose nodes with signals that aren't the parent node.

**Reply by ghost** [reply]
Especially when you have dozens of instances of the same node and need manual set up each children which leads to overwhelming nodes doc. As one additional and very useful use case can mention something like ECS https://www.youtube.com/watch?v=rCu8vQrdDDI
